### ISSUE TITLE
superfluous_disable_command warning displayed when disable opening_brace warning.

### ISSUE THREAD

**Issue by r-plus** [realm/SwiftLint#3514]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

for this file, SwiftLint will warn line 3 for opening_brace rule.

```swift
if true ||
    false
{
    ...
}
```

```
$ swiftlint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
/tmp/work/test.swift:3:1: warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

So I added disable comment for it.

```
if true ||
    false
// swiftlint:disable:next opening_brace
{
    ...
}
```

then SwiftLint warn superfluous_disable_command instead.

```
$ swiftlint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
/tmp/work/test.swift:4:1: warning: Superfluous Disable Command Violation: SwiftLint rule 'opening_brace' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
Done linting! Found 1 violation, 0 serious in 1 file.
```

##### Complete output when running SwiftLint, including the stack trace and command used

already described in adove.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.42.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file: Nothing
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 12.3
* Do you have a sample that shows the issue? `echo "if true ||\n    false\n// swiftlint:disable:next opening_brace\n{\n    ...\n}" | swiftlint`

when fixing #3476, This superfluous_disable_command warning is correct.

**Comment by jpsim** [comment]
The issue is that the presence of the comment in that location makes the violation go away.

```swift
if true ||
    false
// any comment
{
    print("hi")
}
```
